feat(`BaseGuildTextChannel`): Add `BaseGuildTextChannel#sendable` getter.

WiseDevHelper opened the pull request:
**Please describe the changes this PR makes and why it should be merged:**

This PR adds a getter which indicates if the client can send messages in it.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

<!--
Please move lines that apply to you out of the comment:
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- This PR **only** includes non-code changes, like changes to documentation, README, etc.
-->

monbrey commented:
Not a fan of this or #6419. Permission helpers don't need to bloat the library imo

ImRodry commented:
> Not a fan of this or #6419. Permission helpers don't need to bloat the library imo

the library is literally full of permission helpers @monbrey, some classes have more than 5, some even close to 10

wasdennnoch commented:
If this is actually implemented, this also has to check for `VIEW_CHANNEL` permissions. Without those, `SEND_MESSAGES` is implicitly denied.

monbrey commented:
> > Not a fan of this or #6419. Permission helpers don't need to bloat the library imo
> 
> the library is literally full of permission helpers @monbrey, some classes have more than 5, some even close to 10

And we have on multiple occassions suggested we don't want to go further in that direction and increase the problem. Unless that's changed and I haven't realised.

ImRodry commented:
> > > Not a fan of this or #6419. Permission helpers don't need to bloat the library imo
> > 
> > 
> > the library is literally full of permission helpers @monbrey, some classes have more than 5, some even close to 10
> 
> And we have on multiple occassions suggested we don't want to go further in that direction and increase the problem. Unless that's changed and I haven't realised.

I haven’t seen any signs of this slowing down because they keep getting introduced as new classes come. However I’m still not a fan of this PR, but I really wanna see the other one merged because it does represent a feature within Discord and it’s something I’ve tried to suggest before

iCrawl commented:
> I haven’t seen any signs of this slowing down because they keep getting introduced as new classes come.

I don't remember merging any significant permission helpers in a long time.

WiseDevHelper commented:
> Not a fan of this or #6419. Permission helpers don't need to bloat the library imo

Added this to maintain consistency with `ThreadChannel`. Alternatively, the getter from that class could be removed.


> If this is actually implemented, this also has to check for `VIEW_CHANNEL` permissions. Without those, `SEND_MESSAGES` is implicitly denied.

Fixed 👍
